File: samples/maui/KsDialogs.Sample.Maui/TransitionDialogViewModel.cs
namespace KsDialogs.Sample.Maui;

/// <summary>
/// Transition Dialog の ViewModel。
/// </summary>
/// <remarks>
/// 結果型は <see cref="bool"/> で、完了操作は <see langword="true"/> を報告する。
/// この型そのものがレジストリの登録キーになる。
/// </remarks>
/// <param name="message">ダイアログに表示するメッセージ。</param>
/// <param name="transition">
/// 中身へ添付する出入りの演出。演出は呼び出しごとに変わるため、
/// 選ばれた組をここに載せて View factory へ運ぶ。
/// </param>
public sealed class TransitionDialogViewModel(string message, DialogTransition transition) : IDialogViewModel<bool>
{
    /// <summary>ダイアログに表示するメッセージ。</summary>
    public string Message { get; } = message;

    /// <summary>中身へ添付する出入りの演出。</summary>
    public DialogTransition Transition { get; } = transition;
}

File: samples/kmp/shared/src/commonMain/kotlin/jp/kamusoft/ksdialogs/samples/kmp/TransitionDialogViewModel.kt
package jp.kamusoft.ksdialogs.samples.kmp

import jp.kamusoft.ksdialogs.kmp.DialogViewModel

/**
 * Transition Dialog の ViewModel。
 *
 * 結果型は [Boolean] で、完了操作は `true` を報告する。
 * 共有コードで定義したこのクラスがそのまま各 OS のレジストリの登録キーになる。
 *
 * @property message ダイアログに表示するメッセージ
 * @property preset 選ばれた演出。演出の組み立ては各 OS の View 定義側で完結するため、
 *   共有コードは選択だけを運び、View factory が作った View への添付として供給する
 * @property durationMilliseconds 片道の時間 (ミリ秒)
 * @property easing 時間に対する進み方
 */
class TransitionDialogViewModel(
    val message: String,
    val preset: SampleTransitionPreset,
    val durationMilliseconds: Int,
    val easing: SampleEasingPreset,
) : DialogViewModel<Boolean>
